lib/phoenix/endpoint/supervisor.ex

```diff
@@ -351,8 +351,8 @@ defmodule Phoenix.Endpoint.Supervisor do
 
     {scheme, port} =
       cond do
-        https -> {"https", https[:port] || 443}
-        http -> {"http", http[:port] || 80}
+        https -> {"https", transport_port(endpoint, https, :https)}
+        http -> {"http", transport_port(endpoint, http, :http)}
```

Comment:
Do we *need* to be so explicit about sorting through the config ourselves? Rather than special casing the times where the port is dynamic, we could just *always* ask the adapter for the port (I'm happy to add `dynamic_port/2` to `Bandit.PhoenixAdapter`) and be done with it. Taken with the changes suggested for domain sockets above, it would strip away most of the complexity of the changes to this file, I think.

Reply:
That sounds good, but it also kinda prepends on the adapter being able to return a port / being started. That should be a simpler check though.

Reply:
That's a good point - at many of the calls we're guaranteed that the server *won't* be running. What ought to be the behaviour in those cases, do you think?